### ISSUE TITLE
Ensure Consistent LinktaFlow ID Handling and Move ID Mapping from Controller to Service

### ISSUE THREAD

**Issue by kinskiwu** [Linkta-org/core#328]
**Related Ticket**
#329 

**Description**
Currently, the ID mapping logic is handled within the controllers. This should be moved to the service layer to maintain a clean separation of concerns and ensure consistent ID mapping across the application.

Additionally, the LinktaFlow ID handling in the backend have inconsistencies, particularly in the ID mapping logic between different service methods. For example, the createLinktaFlow and fetchLinktaFlow methods have inconsistent ID mappings which can lead to bugs and maintenance challenges.

Examples of Inconsistencies:

Create LinktaFlow:
```typescript
const mappedNodes = nodes.map((node: CustomNode) => ({
  ...node._doc,
  id: node._id,
  _id: undefined,
}));

```
Fetch LinktaFlow:
```typescript
const mappedNodes = nodes.map((node: CustomNode) => ({
  ...node._doc,
  id: node.id,
}));
```

**Objective**
- Ensure consistent ID mapping logic within the backend services.
- Standardize the ID mapping logic in the service layer.

**Tasks**
Refactor Service:
- Move the ID mapping logic from the controllers to the corresponding services.
- Ensure the relevant service methods perform the mapping and return the mapped data.
- Ensure the ID mapping logic for linktaFlow is consistent in the applicaiton

Update the controllers:
- Remove the ID mapping logic from the controller.
- Ensure the controller calls the refactored service method and handles the response accordingly.

**Acceptance Criteria**
- ID mapping logic is handled within the service layer
- Controller fetches already mapped data from the service.
- Consistent ID mapping across the application.
- Proper error handling is in place for all operations.

**Comment by kinskiwu** [comment]
This issue is partially resolved in #364 after removing id mapping from generateLinktaFlowfromUserInput

**Comment by kinskiwu** [comment]
Closing this ticket because it is no longer uncessary
